test/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeServiceSpec.scala

```diff
@@ -0,0 +1,207 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.service
+
+import org.mockito.Matchers
+import org.mockito.Mockito.{times, verify, when}
+import org.mockito.Matchers.any
+import org.scalatest.mock.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import uk.gov.hmrc.domain.{Generator, Nino}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.cache.UpdateNextYearsIncomeCacheModel
+import uk.gov.hmrc.tai.model.domain.income._
+import uk.gov.hmrc.tai.model.domain.{Employment, EmploymentIncome}
+import uk.gov.hmrc.tai.util.constants.journeyCache.UpdateNextYearsIncomeConstants
+import uk.gov.hmrc.time.TaxYearResolver
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+
+class UpdateNextYearsIncomeServiceSpec extends PlaySpec with MockitoSugar {
+
+  "setup" must {
+    "initialize the journey cache and return the cache model" when {
+      "an taxCodeIncome and Employment is returned" in {
+        val nino = generateNino
+
+        when(updateNextYearsIncomeService.employmentService.employment(Matchers.eq(nino), Matchers.eq(employmentId))(any()))
+          .thenReturn(Future.successful(Some(employment(employmentName))))
+
+        when(updateNextYearsIncomeService.taxAccountService.taxCodeIncomeForEmployment(
+          Matchers.eq(nino), Matchers.eq(TaxYear()), Matchers.eq(employmentId))(any())
+        ).thenReturn(Future.successful(Some(taxCodeIncome(employmentName, employmentId, employmentAmount))))
+
+        val result = Await.result(updateNextYearsIncomeService.setup(employmentId, nino), 5.seconds)
+
+        verify(updateNextYearsIncomeService.journeyCacheService, times(1))
+          .cache(expectedMap(employmentName, employmentId, employmentAmount))
+
+        result mustBe UpdateNextYearsIncomeCacheModel(employmentName, employmentId, employmentAmount, None)
+      }
+    }
+
+    "throw a runtime exception" when {
+      "could not retrieve a TaxCodeIncome" in {
+        val nino = generateNino
+
+        when(updateNextYearsIncomeService.employmentService.employment(Matchers.eq(nino), Matchers.eq(employmentId))(any()))
+          .thenReturn(Future.successful(Some(employment(employmentName))))
+
+        when(updateNextYearsIncomeService.taxAccountService.taxCodeIncomeForEmployment(
+          Matchers.eq(nino), Matchers.eq(TaxYear()), Matchers.eq(employmentId))(any())
+        ).thenReturn(Future.successful(None))
+
+        val result: RuntimeException = the[RuntimeException] thrownBy Await.result(updateNextYearsIncomeService.setup(employmentId, nino), 5 seconds)
```

Comment:
I believe that the `5 seconds` causes some warnings about using postfix operators so we might want to use the infix instead.

Reply:
Also we might want to use this https://hmrcdigital.slack.com/archives/GAU3ZD7PG/p1541607006018700